### PR TITLE
refactor: convert macros to all-caps

### LINKED ISSUE #6297: macros all-caps refactor
Recently Bram issued a number of patches which altered how various macros are named. There is a point regarding why this is needed: macros behave different from functions. Yet with Neovim codebase this would be rather painful to merge (unless you want to use `sed -r -i -e … src/**/*.[ch]` and ignore all style errors). Thus suggestion:

- Do not import most of these changes, keep lowercase names.
- Yet do not keep macros with lowercase names: if there is an easy way to transform macros into `static inline REAL_FATTR_ALWAYS_INLINE` function, do this instead. Some macros have other variants, see exceptions section below.

# Benefits

- Functions are generally easier to work with (specifically debug).
- Functions allow giving developers some hints (I speak about `const` and attributes). Compiler can deduce most of them itself, but for developers I think it is better to be explicit.
- Many changes to `*.c` files are simply not needed.
- Functions are more robust: if you have macros like `foo(a) (bar(a, a))` you can’t just write `foo(a++)`: this will both increment `a` twice and provoke undefined behaviour. Functions have neither of these problems: `foo(a++)` with function is like `(a++, foo(a - 1, a - 1))` with the example macros.

# Problems

- Functions are more verbose. Especially if you want to attach ALWAYS_INLINE attribute: with the current status it is either needed to alter declarations generator or write all functions with explicit declarations like

    ```C
    static inline foo(const int bar)
      REAL_FATTR_ALWAYS_INLINE REAL_FATTR_PURE REAL_FATTR_WARN_UNUSED_RESULT;
    static inline foo(const int bar)
    {
        return foo_list[bar];
    }
    ```

    Compare with

    ```C
    #define foo(bar) foo_list[bar]
    ```
- Using functions may require type casts where previously there were no.
- Sometimes using function is going to require reordering of includes or definitions/declarations in the header.
- There are still some exceptions that need to be dealt with separately.

# Exceptions

Some macros still cannot be refactored with new approach and will need to be turned into all-caps or refactored the other way:

- Macros which are used for lvalues cannot be expressed easily with functions, and in any case all occurrences of new function lvalue needs to be refactored: where `foo(a) = x` with macros worked with function it would be `*(foo(a)) = x` at best. So no need to bother with functions, just  make it all-caps.
- Macros whose main or one of the main purposes is type casting its argument(s).
- Macros with `#` (tostring) and `##` (concat) operators.
- Macros which yield computed values (e.g. `#define foo ((char[]) { *p_bar, 0 })`). Though it sometimes makes sense to still turn them into functions, but all occurrences will need to be refactored (`foo` to `foo()`). If value is not computed it may be good to consider replacing it with a global `const` variable.
- Macros which implicitly use local variables. Though if you find these in a header and not somewhere around the function they are used in then it is better to refactor them into something which receives local variables explicitly, in an argument.

---

PS: If somebody takes the issue, it is OK to alter macros which were not altered by Bram, including those which do not exist in Vim at all. E.g. `api_set_error` is a macros which falls into “macros with concat operator” exception and needs to be all-caps then.

Marking this entry-level because refactoring is going to be simple, yet time-consuming.

### PR DESCRIPTION
**Pull request by dundargoc**
Closes https://github.com/neovim/neovim/issues/6297

### CODE REVIEW

**Comment by lewis6991** [comment]
Is there any way we could enforce this with clint?

**Comment by dundargoc** [comment]
`clang-tidy` can already do that automatically.

**Comment by dundargoc** [comment]
Also don't really see the reason why we should bother allcapping macros, I mostly did this to close the issue.

File: src/nvim/event/multiqueue.h [hunk 1]
```diff
@@ -9,7 +9,7 @@
 typedef struct multiqueue MultiQueue;
 typedef void (*PutCallback)(MultiQueue *multiq, void *data);
 
-#define multiqueue_put(q, h, ...) \
+#define MULTIQUEUE_PUT(q, h, ...) \
```
Comment (bfredl) [inline src/nvim/event/multiqueue.h:12]: I'd disagree with this. it is an uninteresting implementation detail that this is a macro.

File: src/nvim/eval/encode.c [hunk 2]
```diff
@@ -29,7 +29,7 @@
 #include "nvim/message.h"
 #include "nvim/vim.h"  // For _()
 
-#define utf_char2len(b) ((size_t)utf_char2len(b))
+#define UTF_CHAR2LEN(b) ((size_t)utf_char2len(b))
```
Comment (bfredl) [inline src/nvim/eval/encode.c:32]: why not just change the single use position instead?

File: src/nvim/ex_eval.c [hunk 3]
```diff
@@ -75,7 +75,7 @@
    || (cstack->cs_idx > 0 \
        && !(cstack->cs_flags[cstack->cs_idx - 1] & CSF_ACTIVE)))
 
-#define discard_pending_return(p) tv_free((typval_T *)(p))
+#define DISCARD_PENDING_RETURN(p) tv_free((typval_T *)(p))
```
Comment (bfredl) [inline src/nvim/ex_eval.c:78]: seems extremely idiosyncratic. just substitute into use place instead?

File: src/nvim/globals.h [hunk 4]
```diff
@@ -104,24 +104,24 @@ EXTERN bool ns_hl_changed INIT(= false);  // highlight need update
 #if (defined(_MSC_VER) && (_MSC_VER >= 1300)) || defined(__MINGW32__)
 typedef __int64 off_T;
 # ifdef __MINGW32__
-#  define vim_lseek lseek64
```
Comment (bfredl) [inline src/nvim/globals.h:107]: these are direct aliases to functions, I don't think these should be uppercase

File: src/nvim/hardcopy.c [hunk 5]
```diff
@@ -1869,7 +1869,7 @@ static void prt_dsc_text(char *comment, char *text)
   prt_write_file(prt_line_buffer);
 }
 
-#define prt_dsc_atend(c)        prt_dsc_text((c), "atend")
+#define PRT_DSC_ATEND(c)        prt_dsc_text((c), "atend")
```
Comment (bfredl) [inline src/nvim/hardcopy.c:1872]: literally a single usecase, just apply it

File: src/nvim/marktree.c [hunk 6]
```diff
@@ -58,7 +58,7 @@
 
 #define ID_INCR (((uint64_t)1) << 2)
 
-#define rawkey(itr) (itr->node->key[itr->i])
+#define RAWKEY(itr) (itr->node->key[itr->i])
```
Comment (bfredl) [inline src/nvim/marktree.c:61]: no.

File: src/nvim/marktree.c [hunk 7]
```diff
@@ -100,20 +100,20 @@ static void compose(mtpos_t *base, mtpos_t val)
 # include "marktree.c.generated.h"
 #endif
 
-#define mt_generic_cmp(a, b) (((b) < (a)) - ((a) < (b)))
+#define MT_GENERIC_CMP(a, b) (((b) < (a)) - ((a) < (b)))
```
Comment (bfredl) [inline src/nvim/marktree.c:103]: no.

File: src/nvim/highlight_defs.h [hunk 8]
```diff
@@ -205,7 +205,7 @@ typedef struct {
   int ns_id;
   int syn_id;
 } ColorKey;
-#define ColorKey(n, s) (ColorKey) { .ns_id = (int)(n), .syn_id = (s) }
```
Comment (bfredl) [inline src/nvim/highlight_defs.h:208]: no.

File: src/nvim/ex_eval.c [hunk 9]
```diff
@@ -75,8 +75,6 @@
    || (cstack->cs_idx > 0 \
        && !(cstack->cs_flags[cstack->cs_idx - 1] & CSF_ACTIVE)))
 
-#define discard_pending_return(p) tv_free((typval_T *)(p))
```
Comment (zeertzjq) [inline src/nvim/ex_eval.c:78]: Removing this looks confusing. How about converting this to a function?

File: src/nvim/vim.h [hunk 10]
```diff
@@ -253,7 +253,7 @@ enum { FOLD_TEXT_LEN = 51, };  //!< buffer size for get_foldtext()
 #define STRNCAT(d, s, n)    strncat((char *)(d), (char *)(s), (size_t)(n))
 #define STRLCAT(d, s, n)    xstrlcat((char *)(d), (char *)(s), (size_t)(n))
 
-#define vim_strpbrk(s, cs) (char_u *)strpbrk((char *)(s), (char *)(cs))
+#define VIM_STRPBRK(s, cs) (char_u *)strpbrk((char *)(s), (char *)(cs))
```
Comment (bfredl) [inline src/nvim/vim.h:256]: maybe we can just use `strpbrk` at usage places? most seem to artifically convert _to_ char_u anyway

**Review by justinmk — APPROVED** [review]

File: src/nvim/ex_eval.c [hunk 11]
```diff
@@ -75,7 +75,10 @@
...
+void discard_pending_return(typval_T *p)
```
Comment (zeertzjq) [inline src/nvim/ex_eval.c:78]:
```suggestion
static void discard_pending_return(typval_T *p)
```

**Pull request merged**